client/request.go

```diff
@@ -305,10 +290,16 @@ func encodeData(data interface{}) (*bytes.Buffer, error) {
 	return params, nil
 }
 
-func ensureReaderClosed(response serverResponse) {
-	if response.body != nil {
+func ensureReaderClosed(response *http.Response) {
+	if response != nil && response.Body != nil {
 		// Drain up to 512 bytes and close the body to let the Transport reuse the connection
-		_, _ = io.CopyN(io.Discard, response.body, 512)
-		_ = response.body.Close()
+		// see https://github.com/google/go-github/pull/317/files#r57536827
+		//
+		// TODO(thaJeztah): see if this optimization is still needed, or already implemented in stdlib,
+		//   and check if context-cancellation should handle this as well. If still needed, consider
+		//   wrapping response.Body, or returning a "closer()" from [Client.sendRequest] and related
+		//   methods.
+		_, _ = io.CopyN(io.Discard, response.Body, 512)
+		_ = response.Body.Close()
```

Comment:
Left this as a TODO, as it needs some investigating what's still needed, and what would be the most suitable approach. All of this is non-exported, so we can change at any moment.